src/plugins/imagery/components/ImageryViewLayout.vue

```diff
@@ -262,6 +331,69 @@ export default {
 
             this.time = this.timeFormat.format(datum);
             this.imageUrl = this.imageFormat.format(datum);
+        },
+        getFormatter(key) {
+            let valueFormatter = this.openmct.telemetry.getValueFormatter({
+                format: key
+            });
+
+            return valueFormatter.formatter;
+        },
+        numericImageAge() {
+            let currentTime = this.currentTimeValue();
+            let selectedImage = this.getSelectedImage();
+            let formatter = this.getFormatter(this.timeKey);
+
+            if (selectedImage === undefined || selectedImage[this.timeKey] === undefined) {
+                return;
+            }
+
+            let parsedSelectedTime = formatter.parse(selectedImage[this.timeKey]);
+
+            return currentTime - parsedSelectedTime;
+        },
+        formattedAge() {
+            let age = this.numericImageAge();
+
+            if (!age) {
+                return;
+            }
+
+            let result = this.durationFormatter.format(age);
+
+            if (age > EIGHT_HOURS) {
+                let negativeAge = (age / ONE_HOUR) * -1;
+                result = moment.duration(negativeAge, 'hours').humanize(true);
+
+                if (age > TWENTYFOUR_HOURS) {
+                    negativeAge = (age / TWENTYFOUR_HOURS) * -1;
+                    result = moment.duration(negativeAge, 'days').humanize(true);
+                }
+            }
+
+            return result;
+        },
+        trackAge() {
+            let age;
+            this.clearAgeTracking();
+            this.ageTracker = window.setInterval(() => {
+                age = this.formattedAge();
```

Comment:
Let's track `numericalAge` on `this` here, and make the formatted age a computed property. That way, `trackAge` is only responsible for tracking the age, and Vue can optimize calls to `formatAge`.

Reply:
numericImageAge computed value has been created